CanHazFunny/CanHazFunny/Jester.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CanHazFunny;
+
+    public class Jester 
+    {   
+        
+        private IJokeService? jokeService { get; set; } // set in constructor
+        private IConsoleOutput? consoleOutput { get; set; }
+
+        public Jester() // for mocks...
+        {
+
+        }
+
+         public Jester(IJokeService IjokeService, IConsoleOutput IconsoleOutput)
+         {
+            if (IconsoleOutput is null || IjokeService is null)
+            {
+                throw new ArgumentNullException("an arg was null");
+            }
+            jokeService = IjokeService;
+            consoleOutput = IconsoleOutput;
+         }
+       
+       
+       
+        public void TellJoke()
+        {
+            string joke = jokeService!.GetJoke(); // ! because TellJoke cant be called without instance, and constructor checks for null
```

Comment:
Implementing my above suggestions will eliminate the need to tell the compiler jokeService is not null.
```suggestion
            string joke = jokeService.GetJoke();
```